### PR TITLE
[TUBEMQ-594] Trpc-go tube sdk strongly rely on local config

### PR DESCRIPTION
**Pull request by gosonzhang**
1. Adjust the StartTubeMQService API to support service initialization according to TubeMQServiceConfig object;
2. Adjust the ToString() function definition of BaseConfig,ConsumerConfig classes

### CODE REVIEW

File: tubemq-client-twins/tubemq-client-cpp/include/tubemq/tubemq_config.h [hunk 1]
```diff
@@ -34,6 +34,42 @@ using std::set;
 using std::string;
 
 
+// configure for log, thread pool etc.
+class TubeMQServiceConfig {
+ public:
+  TubeMQServiceConfig();
+  ~TubeMQServiceConfig();
+  TubeMQServiceConfig& operator=(const TubeMQServiceConfig& target);
+  void setLogCofigInfo(int32_t log_max_num,
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-cpp/include/tubemq/tubemq_config.h:43]: The public function first letter needs to be capitalized.

Comment (gosonzhang) [reply tubemq-client-twins/tubemq-client-cpp/include/tubemq/tubemq_config.h:43]: Ok

File: tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc [hunk 2]
```diff
@@ -32,6 +32,112 @@ using std::set;
...
+  timer_threads_ = tb_config::kTimerThreadNumDef;
+  network_threads_ = tb_config::kNetworkThreadNumDef;
+  signal_threads_ = tb_config::kSignalThreadNumDef;
+}
+
+TubeMQServiceConfig::~TubeMQServiceConfig() {
+  //
+}
+
+TubeMQServiceConfig& TubeMQServiceConfig::operator=(const TubeMQServiceConfig& target) {
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc:52]: Just use the default function.

File: tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc [hunk 3]
```diff
@@ -32,6 +32,112 @@ using std::set;
 using std::stringstream;
 using std::vector;
 
+
+
+TubeMQServiceConfig::TubeMQServiceConfig() {
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc:37]: It is better to use the initialization list for initialization.

File: tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc (hunk 2 above)
Comment (gosonzhang) [reply tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc:52]: Ok

File: tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc (hunk 3 above)
Comment (gosonzhang) [reply tubemq-client-twins/tubemq-client-cpp/src/tubemq_config.cc:37]: I think it’s better to keep the current initialization mode: if the initialization is done in the .h file, the internal const_config.h file will be exposed to the outside, which will make the API definitions of the user contact not concise enough

**Comment by charlely** [comment]
LGTM

**Comment by gosonzhang** [comment]
Thanks @charlely

**Pull request merged**